File: OS/Experiment 3/FCFS.c
// write a code for FCFS scheduling algorithm
#include<stdio.h>
int main()
{
    int n, i, temp, sum = 0, wait = 0, intTat = 0;
    float avg_wait, avg_tat;
    printf("Enter the number of processes: ");
    scanf("%d", &n);
    int bt[n], wt[n], tat[n];
    printf("Enter the burst time of the processes: \n");
    for(i = 0; i < n; i++)
    {
        scanf("%d", &bt[i]);
    }
    wt[0] = 0;
    for(i = 1; i < n; i++)
    {
        wt[i] = wt[i-1] + bt[i-1];
    }
    for(i = 0; i < n; i++)
    {
        tat[i] = wt[i] + bt[i];
    }
    for(i = 0; i < n; i++)
    {
        wait = wait + wt[i];
        intTat = intTat + tat[i];
    }
    avg_wait = (float)wait/n;
    avg_tat = (float)intTat/n;
    printf("Process\tBurst Time\tWaiting Time\tTurn Around Time \n");
    for(i = 0; i < n; i++)
    {
        printf("%d\t%d\t\t%d\t\t%d \n", i+1, bt[i], wt[i], tat[i]);
    }
    printf("Average waiting time: %f \n", avg_wait);
    printf("Average turn around time: %f \n", avg_tat);
    return 0;
}

File: OS/Experiment 3/fcfs.py
import time

def FCFS():
    waitingTime.append(0)
    completionTime.append(arrivalTime[0] + burstTime[0])
    turnAroundTime.append(completionTime[0] - arrivalTime[0])
    # used to calculate the waitingTime, completionTime and turnAroundTime for each process
    for i in range(1, numberOfProcess):
        processWaitingTime = completionTime[i-1] - arrivalTime[i]
        waitingTime.append(processWaitingTime)
        
        processCompletionTime = completionTime[i-1] + burstTime[i]
        completionTime.append(processCompletionTime)
        
        processTurnAroundTime = completionTime[i] - arrivalTime[i]
        turnAroundTime.append(processTurnAroundTime)
    

    averageWaitingTime = sum(waitingTime) / numberOfProcess
    averageTurnAroundTime = sum(turnAroundTime) / numberOfProcess
    print("Process \t Arrival Time \t Burst Time  \t Waiting Time \t Completion Time \t Turn Around Time")
    for i in range(numberOfProcess):
        print(i+1, "\t\t", arrivalTime[i], "\t\t", burstTime[i], "\t\t", waitingTime[i], "\t\t", completionTime[i], "\t\t\t", turnAroundTime[i])

    print("\nAverage Waiting Time: ", averageWaitingTime)
    print("Average Turn Around Time: ", averageTurnAroundTime)


numberOfProcess = int(input("Enter the number of processes : "))
arrivalTime, burstTime, completionTime, waitingTime, turnAroundTime  = [], [], [], [], []
arrivalTime=[0,1,2,3,6]
burstTime=[2,6,4,9,12]
# for i in range(numberOfProcess):
#     processArrivalTime, processBurstTime = int(input("Enter the arrival time and burst time of process {} : ".format(i+1)))
#     arrivalTime.append(processArrivalTime)
#     burstTime.append(processBurstTime)

t1=time.time()
FCFS()
t2=time.time()
print("Time taken by FCFS execution : ",t2-t1, " sec")
